app/src/main/java/com/lastaoutdoor/lasta/ui/screen/settings/SettingsComponent.kt

```diff
@@ -31,131 +43,218 @@ fun SettingsComponent(
-              ActivityType.BIKING -> "settingsBiking"
-            }
+        val tag = "settings${activity.name}"
         Button(
             modifier = Modifier.padding(1.dp).testTag(tag),
-            onClick = {
-              selectedActivity = activity
-              updatePrefActivity(activity)
-            },
+            onClick = { updatePrefActivity(activity) },
             colors =
                 ButtonDefaults.buttonColors(
                     containerColor =
-                        if (selectedActivity == activity) MaterialTheme.colorScheme.primary
+                        if (prefActivity == activity) MaterialTheme.colorScheme.primary
                         else MaterialTheme.colorScheme.secondaryContainer,
                     contentColor = MaterialTheme.colorScheme.onPrimary)) {
-              val sport =
-                  when (activity) {
-                    ActivityType.CLIMBING -> LocalContext.current.getString(R.string.climbing)
-                    ActivityType.HIKING -> LocalContext.current.getString(R.string.hiking)
-                    ActivityType.BIKING -> LocalContext.current.getString(R.string.biking)
-                  }
-              Text(text = sport)
+              Text(text = activity.resourcesToString(LocalContext.current))
             }
       }
     }
   }
+}
 
-  Row {
-    Column(horizontalAlignment = Alignment.CenterHorizontally) {
-      for (activity in ActivityType.values()) {
-        Text(
-            text = activity.resourcesToString(LocalContext.current),
-            color = MaterialTheme.colorScheme.primary,
-            fontWeight = FontWeight.Bold,
-            style = MaterialTheme.typography.headlineMedium)
-
-        Spacer(modifier = Modifier.height(8.dp))
-
-        DropDownMenuComponent(
-            items = UserLevel.values().toList(),
-            selectedItem =
-                when (activity) {
-                  ActivityType.CLIMBING -> climbingLevel
-                  ActivityType.HIKING -> hikingLevel
-                  ActivityType.BIKING -> bikingLevel
-                },
-            onItemSelected = { newLevel: UserLevel ->
-              when (activity) {
-                ActivityType.CLIMBING -> {
-                  climbingLevel = newLevel
-                  updateClimbingLevel(newLevel)
-                }
-                ActivityType.HIKING -> {
-                  hikingLevel = newLevel
-                  updateHikingLevel(newLevel)
-                }
-                ActivityType.BIKING -> {
-                  bikingLevel = newLevel
-                  updateBikingLevel(newLevel)
+@Composable
+fun ActivityLevelsComponent(
+    levels: UserActivitiesLevel,
+    updateClimbingLevel: (UserLevel) -> Unit,
+    updateHikingLevel: (UserLevel) -> Unit,
+    updateBikingLevel: (UserLevel) -> Unit,
+) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = LocalContext.current.getString(R.string.select_lev_activity),
```

Comment:
Also, for a homegenous look, try to use the same kind of text as for the FilterScreen: 
```suggestion
           Text(
                text = stringResource(id = R.string.filter_activity_type),
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold)
```

Reply:
okay will do :)